### PR TITLE
Allow the `name_scheme` argument to `python_wheel` to be a list instead of a single string.

### PR DESCRIPTION
**Pull request by toastwaffle**

### CODE REVIEW

**Review by goddenrich — APPROVED** [review]
Im not sure if this is a general thing with github but I can seem to add a comment to build_def/python.build_defs line 515:522 which says 
```

    This is a lightweight pip-free alternative to pip_library which supports cross-compiling.
    Rather than leaning on pip which is difficult to achieve reproducible builds with and
    support on different platforms, this rule is a simple wrapper around curl and unzip.
    Unless otherwise specified, the wheels are expected to adhere to common naming schemes,
    such as:
      <package_name>-<version>[-<os>-<arch>].whl
      <package_name>-<version>[-<os>_<arch>].whl
      <package_name>-<version>.whl
```
I think this doc needs updating now the wheel resolver can be used

**Comment by tm-jdelapuente** [comment]
Hey, can you provide a bit of context here? I'm not following the motivation to do this change

**Review by tm-jdelapuente — APPROVED** [review]

File: build_defs/python.build_defs [hunk 1]
```diff
@@ -558,30 +571,15 @@ def python_wheel(name:str, version:str, labels:list=[], hashes:list=None, packag
...
-                                                                      version=version)]
+        if not name_scheme:
+            name_scheme = _DEFAULT_WHEEL_NAME_SCHEMES
+        elif isinstance(name_scheme, str):
+            name_scheme = [name_scheme]
+
+        urls += [
+            scheme.format(url_base=url_base, package_name=package_name, initial=initial, version=version)
+            for scheme in name_scheme
+        ]
```
Comment (SpangleLabs) [inline build_defs/python.build_defs:582]: This is all so much more readable, thank you a bunch for that!

File: test/name_scheme_test.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    # The click python_wheel has no name_scheme argument, and is thus using the default value which
+    # is a list.
+    def test_click_is_importable(self):
+        import click
+        self.assertIsNotNone(click.command)
+
+    # The click-log python_wheel has an explicit name_scheme argument which is a string.
+    def test_click_log_is_importable(self):
+        import click_log
+        self.assertIsNotNone(click_log.basic_config)
```
Comment (SpangleLabs) [inline test/name_scheme_test.py:15]: Is there risk of these drifting, as they're not linked to the python_wheel defs in any particular way? So if the def changes, these tests might not still be covering what they claim to be?

**Review by SpangleLabs — APPROVED** [review]

**Review by SpangleLabs — APPROVED** [review]

File: test/name_scheme_test.py (hunk 2 above)
Comment (toastwaffle) [reply test/name_scheme_test.py:15]: Yes, there's a risk. This isn't the only example of a test relying on a package in //third_party/python, but I'm not sure there's an alternative. I've added a comment on the packages in question to draw attention to this test.

**Pull request merged**